### PR TITLE
[FIX] Download on clicking start button

### PR DESCRIPTION
**Pull request by hv7214**
## Description

- Added code to append `handler` to `handlerLst`.
- Added log for socket close.
- Added `Start` and `Kill` endpoint's method (**GET**).

## Related Issue

#806 
## Motivation and Context

Empty handler list leads to infinite loop which in turn disable the download. In started function a handler is created which should be appended to `handlerLst`.

## How Has This Been Tested?

Now files are successfully downloaded requested by any user. Pie chart screenshot: 
Screenshot from 2020-01-31 15-28-01
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by carpecodeum** [comment]
@hv7214 hey as you might have noticed I was assigned to this issue and was currently working on it. Only if you could have asked me before working on this issue. Never mind if you could follow this in the future it would be better and more professional. Happy Coding :)

**Comment by hv7214** [comment]
@carpecodeum i noticed it's been a month since you were assigned. So i asked @kmehant about the status of same. He said if you haven't made any PR then i can work on the issue. And by the way it's more professional not to take up every issue and put it on hold. And sorry if i offended you :)

**Comment by carpecodeum** [comment]
> @carpecodeum i noticed it's been a month since you were assigned. So i asked @kmehant about the status of same. He said if you haven't made any PR then i can work on the issue. And by the way it's more professional not to take up every issue and put it on hold. And sorry if i offended you :)

Oh, I was finding a little trouble finding a solution for this issue and no I haven't taken every issue I am currently working on only two issues so there are many more left unattended! And this was the only issue which was put on hold due to the difficulty I faced the rest of the issues I am working on very actively and contributing. I am more than happy that this issue got resolved it is for the greater good of this project I was just saying you could have at least commented on this issue before starting to work on it I would have unassigned myself and it would have been more friendly in the world of open source contribution! Anyways that's my outlook only!

**Comment by kmehant** [comment]
@carpecodeum I agree with you though the final motive is to get the things resolved. I appreciate your time you have put into I request your review from your side as you have already made your hands dirty for this issue :). On a side note you can create a draft pull request with minimal work as you go on solving any issue to project your progress ;).

@hv7214 thank you for the PR and I appreciate it :)

**Comment by carpecodeum** [comment]
@kmehant sure I completely understand your point and yes I will review this Pull Request. I didnt know we could make a draft PR from the next time I'll be sure to do that.

File: components/core/DownloadDaemon.py [hunk 1]
```diff
@@ -240,7 +239,7 @@ def on_error(ws, error):
 
 
 def on_close(ws):
-    pass
+    print("socket closed")
```
Comment (JaDogg) [inline components/core/DownloadDaemon.py]: Do we have a logger? Probably replace `print` with logging

Comment (hv7214) [reply components/core/DownloadDaemon.py]: done

**Comment by kmehant** [comment]
@hv7214 Can you resolve the conflicts

This is a required change :)

**Comment by JaDogg** [comment]
Please resolve the conflicts as Mehant mentioned :)

**Comment by hv7214** [comment]
Sure : )

**Pull request merged**